docs/content/dagster-cloud/deployment/hybrid.mdx

```diff
@@ -1,7 +1,74 @@
 ---
 
 # Hybrid deployment in Dagster Cloud
+
+<Note>This guide is applicable to Dagster Cloud.</Note>
+
+In this guide, we'll cover how Hybrid deployment works in Dagster Cloud, including its system architecture, agent, and security.
+
+---
+
+## Hybrid architecture overview
+
+A **hybrid deployment** utilizes a combination of your infrastructure and Dagster-hosted backend services.
+
+The Dagster backend services - including the web frontend, GraphQL API, metadata database, and daemons (responsible for executing schedules and sensors) - are hosted in Dagster Cloud. You are responsible for running an [agent](/dagster-cloud/deployment/agents) in your environment.
+
+<Image
+alt="Dagster Cloud Architecture Diagram"
+src="/images/dagster-cloud/deployment/hybrid-architecture.png"
+width={1583}
+height={765}
+/>
+
+When users interact with the web frontend, queries are made against the GraphQL API, or schedules and sensors tick. This enqueues work for your agent. The agent polls the agent API to see if any work needs to be done and launches user code as appropriate to fulfill requests. User code then streams metadata back to the agent API (GraphQL over HTTPS) to make it available in Dagster Cloud.
+
+All user code runs within your environment, in isolation from Dagster system code.
+
+---
+
+## The agent
+
+Because the agent communicates with the Dagster Cloud control plane over the agent API, it’s possible to support agents that operate in arbitrary compute environments.
+
+This means that over time, Dagster Cloud’s support for different user deployment environments will expand and custom agents can take advantage of bespoke compute environments such as HPC.
+
+Refer to the [Agents documentation](/dagster-cloud/deployment/agents) for more info, including the agents that are currently supported.
+
+---
+
+## Security
+
+- [User code](#user-code)
+- [Runs](#runs)
+- [Ingress](#ingress)
+
+### User code
+
+When Dagster Cloud needs to interact with user code - for instance, to display the structure of a job in the Dagster Cloud user interface, to run the body of a sensor definition, or to launch a run for a job - it enqueues a message for the Dagster Cloud Agent. The Dagster Cloud Agent picks up this message and then launches or queries user code running on the appropriate compute substrate.
+
+Depending on the agent implementation, user code may run in isolated OS processes, in Docker containers, in ECS Tasks, in Kubernetes Jobs and Services, or in a custom isolation strategy.
+
+Queries to user code run over a well-defined grpc interface. Dagster Cloud uses this interface to:
+
+- Retrieve the names, config schemas, descriptions, tags, and structures of jobs, ops, repositories, partitions, schedules, and sensors defined in your code
+- Evaluate schedule and sensor ticks and determine whether a run should be launched
+
+When the agent queries user code, it writes the response back to Dagster Cloud over a well-defined GraphQL interface.
+
+### Runs
```

Comment:
runs are also considered User Code. Another option instead of breaking up these sections would be to lead with some highlights

- There is no ingress required from Dagster Cloud to user environments
- Dagster Cloud does not have access to user code. It fetches metadata about the code over constrained APIs

(only apply to hybrid)